quill-sql/src/main/scala/io/getquill/source/sql/norm/MergeSecondaryJoin.scala

```diff
@@ -0,0 +1,30 @@
+package io.getquill.source.sql.norm
+
+import io.getquill.ast._
+
+object MergeSecondaryJoin extends StatelessTransformer {
```

Comment:
It doesn't directly relate to `ExpandJoin`. The merge operation is for lifting secondary join AST flatMap/map calls into nested tuple join format.

For example, after `ExpandJoin` and `Normalize` have been called in `Prepare.scala`, given a query like:
```
for{
  (a,b) <- TableA.join.TableB.on(_.id == _.akey)
     c  <- TableC.join(_.id == b.id)
} yield (a,b,c)

```
The AST will look something like:

```
query[TableA].join(query[TableB]).on((a,b)=> a.id == b.akey).
flatMap(...=> c.id == b.id).map(....)
```

We need to lift the flatMap call into the primary join so as to chain the join calls just as if we had used nested tuple join syntax.

End result will look like: 
```
query[TableA].join(query[TableB]).on((a,b)=> a.id == b.akey).
join(query[TableC]).on(c=> c.id == b.id).map(...)